src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java

```diff
@@ -107,23 +106,25 @@ protected boolean preserveClusterUponCompletion() {
     }
 
     @AfterClass
-    public static void dumpCoverage() throws IOException, MalformedObjectNameException {
+    public static void dumpCoverage() {
         // jacoco.dir is set in sqlplugin-coverage.gradle, if it doesn't exist we don't
         // want to collect coverage so we can return early
         String jacocoBuildPath = System.getProperty("jacoco.dir");
-        if (jacocoBuildPath.isEmpty()) {
+        if (jacocoBuildPath == null || jacocoBuildPath.isEmpty()) {
             return;
         }
 
         String serverUrl = "service:jmx:rmi:///jndi/rmi://127.0.0.1:7777/jmxrmi";
-        JMXConnector connector = JMXConnectorFactory.connect(new JMXServiceURL(serverUrl));
-        IProxy proxy = MBeanServerInvocationHandler.newProxyInstance(
-                connector.getMBeanServerConnection(), new ObjectName("org.jacoco:type=Runtime"), IProxy.class,
-                false);
-
-        Path path = Paths.get(jacocoBuildPath + "/integTest.exec");
-        Files.write(path, proxy.getExecutionData(false));
-        connector.close();
+        try(JMXConnector connector = JMXConnectorFactory.connect(new JMXServiceURL(serverUrl))) {
+            IProxy proxy = MBeanServerInvocationHandler.newProxyInstance(
+                    connector.getMBeanServerConnection(), new ObjectName("org.jacoco:type=Runtime"), IProxy.class,
+                    false);
+
+            Path path = Paths.get(jacocoBuildPath + "/integTest.exec");
+            Files.write(path, proxy.getExecutionData(false));
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to dump coverage: " + ex);
```

Comment:
Would be better if we keep the full error stack by `throw new RuntimeException("Failed to dump coverage", ex);`

Reply:
Agreed.